soroban-sdk-macros/src/derive_event.rs

```diff
@@ -244,7 +250,14 @@ fn derive_impls(args: &ContractEventArgs, input: &DeriveInput) -> Result<TokenSt
         },
         DataFormat::Vec => quote! {
             use #path::IntoVal;
-            (#(&self.#data_idents,)*).into_val(env)
+            (
+                #(
+                    {
+                        let val: #path::Val = self.#data_idents.into_val(env);
```

Comment:
I changed this conversion to preconvert to a `Val`, so that the macro code didn't need to handle the case where the field is a ref or not, which would otherwise complicate it. There should be no harm in doing this.